filament/src/MaterialInstance.cpp

```diff
@@ -177,24 +177,39 @@ template UTILS_PUBLIC void MaterialInstance::setParameter<mat4f>   (const char*
 
 FMaterialInstance::FMaterialInstance() noexcept = default;
 
-FMaterialInstance::FMaterialInstance(FEngine& engine, FMaterial const* material, const char* name) :
-        mName(name) {
+FMaterialInstance::FMaterialInstance(FEngine& engine,
+        FMaterialInstance const* other, const char* name)
+        : mPolygonOffset(other->mPolygonOffset),
+          mCulling(other->mCulling),
+          mColorWrite(other->mColorWrite),
+          mDepthWrite(other->mDepthWrite),
+          mDepthFunc(other->mDepthFunc),
+          mScissorRect(other->mScissorRect),
+          mName(name ? CString(name) : other->mName) {
```

Comment:
Will these get clobbered later when `initialize()` is called?

Reply:
So I changed initialize() so that doesn't happen.